20250117-actors-pubsub.md

```diff
@@ -0,0 +1,140 @@
+# Actor PubSub Implementation Proposal
+
+- Author(s): @joshvanl
+
+## Overview
+
+This proposal introduces PubSub capabilities for Dapr Actors, enabling actors to subscribe to messages within their type for a particular topic.
+Additionally, actors of a specific type will have the ability to publish messages to a topic.
+A single designated PubSub component will be marked as the actor PubSub, similar to how an actor state store is defined.
+There can be only one actor PubSub state store per Dapr deployment.
+
+## Background
+
+Currently, Dapr provides PubSub functionality for applications but lacks native PubSub integration for actors.
+This limitation means that actors cannot communicate efficiently via publish-subscribe messaging patterns within their type.
+By enabling Actor PubSub, actors can directly subscribe to and publish messages to specific topics, improving scalability and communication efficiency.
+
+## Implementation Details
+
+### Actor PubSub Registration
+
+A single PubSub component will be designated as the actor PubSub, similar to how an actor state store is configured.
+The metadata configuration will follow the existing pattern:
+
+```yaml
+metadata:
+  - name: actorpubsub
+    value: "true"
+```
+
+Only one PubSub component can be marked as the actor PubSub.
+
+### Actor Subscriptions
+
+Actors will be able to subscribe to messages within their type for a particular topic.
```

Comment:
I'd like to see this supplemented with support for [content-based routing](https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-route-messages/). While a general broadcast is certainly a much-needed feature, I can easily imagine a chatty topic activating actors unnecessarily. Limiting such activations to only those actors registered to receive that subset would be a solid (and likely expected) part of this functionality.